### PR TITLE
Update procstat to support cgroup globs & include systemd unit children (Copy of #7890)

### PR DESCRIPTION
**Pull request by sspaink**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

Thank you @DEvil0000 for the work on the original pull request (#7890)! I will be taking it over to help resolve merge conflicts and any other required changes.

Closing #7890 and moving forward with this new pr.

### CODE REVIEW

File: plugins/inputs/procstat/procstat.go [hunk 1]
```diff
@@ -56,9 +58,10 @@ var sampleConfig = `
   # pattern = "nginx"
   ## user as argument for pgrep (ie, pgrep -u <user>)
   # user = "nginx"
-  ## Systemd unit name
+  ## Systemd unit name, supports globs when include_systemd_children is set to true
```
Comment (ssoroka) [inline plugins/inputs/procstat/procstat.go:61]: these comments don't match the readme

File: plugins/inputs/procstat/procstat.go [hunk 2]
```diff
@@ -116,33 +125,48 @@ func (p *Procstat) Gather(acc telegraf.Accumulator) error {
...
+			acc.AddFields("procstat_lookup", fields, tags, now)
+			return err
 		}
-		tags := map[string]string{
-			"pid_finder": p.PidFinder,
-			"result":     "lookup_error",
+
+		err = p.updateProcesses(pids, tags, p.procs, newProcs)
+		if err != nil {
+			acc.AddError(fmt.Errorf("E! Error: procstat getting process, exe: [%s] pidfile: [%s] pattern: [%s] user: [%s] %s",
```
Comment (ssoroka) [inline plugins/inputs/procstat/procstat.go:153]:
```suggestion
			acc.AddError(fmt.Errorf("procstat getting process, exe: [%s] pidfile: [%s] pattern: [%s] user: [%s] %s",
```

File: plugins/inputs/procstat/procstat.go [hunk 3]
```diff
@@ -367,16 +388,34 @@ func (p *Procstat) getPIDFinder() (PIDFinder, error) {
...
-	var pids []PID
-	tags := make(map[string]string)
-	var err error
+func (p *Procstat) findPids() []PidsTags {
+	var pidTags []PidsTags
 
 	f, err := p.getPIDFinder()
 	if err != nil {
-		return nil, nil, err
+		pidTags = append(pidTags, PidsTags{nil, nil, err})
```
Comment (ssoroka) [inline plugins/inputs/procstat/procstat.go:396]: I assume it's intentional that we're continuing after the error?

**Review by ssoroka — CHANGES_REQUESTED** [review]
minor comments

File: plugins/inputs/procstat/procstat.go (hunk 3 above)
Comment (sspaink) [reply plugins/inputs/procstat/procstat.go]: Good catch, it wasn't intentional. I moved the `p.getPIDFinder` call closer to where the returned value is being used and return if an error is found.

**Review by helenosheaa — APPROVED** [review]
LGTM

File: plugins/inputs/procstat/README.md [hunk 4]
```diff
@@ -26,8 +26,9 @@ Processes can be selected for monitoring using one of several methods:
   # pattern = "nginx"
   ## user as argument for pgrep (ie, pgrep -u <user>)
   # user = "nginx"
-  ## Systemd unit name
+  ## Systemd unit name, supports globs when include_systemd_children is set to true
```
Comment (reimda) [inline plugins/inputs/procstat/README.md:29]: Hi Sebastian, the sample config in procstat.go has include_systemd_children commented out with its default value. It doesn't show up here in the readme. Could you make them match in a new PR?

Comment (sspaink) [reply plugins/inputs/procstat/README.md:29]: Good catch, new pr to fix it: https://github.com/influxdata/telegraf/pull/9553

**Pull request merged**